src/sinks/splunk_hec/common/acknowledgements.rs

```diff
@@ -0,0 +1,314 @@
+            *retries = retries.checked_sub(1).unwrap_or(0);
+        }
+    }
+
+    /// Removes all expired ack ids (those with a retry count of 0) and
+    /// finalizes associated events with the given status
+    fn expire_ack_ids_with_status(&mut self, status: EventStatus) {
+        let expired_ack_ids = self
+            .acks
+            .iter()
+            .filter_map(|(ack_id, (retries, _))| (*retries == 0).then(|| *ack_id))
+            .collect::<Vec<_>>();
+        for ack_id in expired_ack_ids {
+            if let Some((_, ack_event_status_sender)) = self.acks.remove(&ack_id) {
+                let _ = ack_event_status_sender.send(status);
+            }
+        }
+    }
+
+    // Sends an ack status query request to Splunk HEC
+    async fn send_ack_query_request(
+        &mut self,
+        request_body: &HecAckStatusRequest,
+    ) -> Result<HecAckStatusResponse, HecAckApiError> {
+        self.decrement_retries();
+        let request_body_bytes =
+            serde_json::to_vec(request_body).map_err(|_| HecAckApiError::ClientBuildRequest)?;
+        let request = self
+            .http_request_builder
+            .build_request(request_body_bytes, "/services/collector/ack")
+            .map_err(|_| HecAckApiError::ClientBuildRequest)?;
+
+        let response = self
+            .client
+            .send(request.map(Body::from))
+            .await
+            .map_err(|_| HecAckApiError::ServerSendQuery)?;
+
+        let status = response.status();
+        if status.is_success() {
+            let response_body = hyper::body::to_bytes(response.into_body())
+                .await
+                .map_err(|_| HecAckApiError::ClientParseResponse)?;
+            serde_json::from_slice::<HecAckStatusResponse>(&response_body)
+                .map_err(|_| HecAckApiError::ClientParseResponse)
+        } else if status.is_client_error() {
+            Err(HecAckApiError::ClientSendQuery)
+        } else {
+            Err(HecAckApiError::ServerSendQuery)
+        }
+    }
+}
+
+pub async fn run_acknowledgements(
+    mut receiver: UnboundedReceiver<(u64, Sender<EventStatus>)>,
+    client: HttpClient,
+    http_request_builder: Arc<HttpRequestBuilder>,
+    indexer_acknowledgements: HecClientAcknowledgementsConfig,
+) {
+    let mut interval = tokio::time::interval(Duration::from_secs(
+        indexer_acknowledgements.query_interval.get() as u64,
+    ));
+    let mut ack_client = HecAckClient::new(
+        indexer_acknowledgements.retry_limit.get(),
+        client,
+        http_request_builder,
+    );
+
+    loop {
+        tokio::select! {
+            _ = interval.tick() => {
+                ack_client.run().await;
+            },
+            ack_info = receiver.recv() => {
+                match ack_info {
+                    Some((ack_id, tx)) => {
+                        ack_client.add(ack_id, tx);
+                        debug!("Stored ack id {}", ack_id);
```

Comment:
```suggestion
                        debug!(message = "Stored ack id", ?ack_id);
```

Same feedback as above about separating the message and fields.